openmc/mgxs/mgxs.py

```diff
@@ -4615,3 +4628,728 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
+        being tracked. This is unity if the by_nuclide attribute is False.
+    nuclides : Iterable of str or 'sum'
+        The optional user-specified nuclides for which to compute cross
+        sections (e.g., 'U-238', 'O-16'). If by_nuclide is True but nuclides
+        are not specified by the user, all nuclides in the spatial domain
+        are included. This attribute is 'sum' if by_nuclide is false.
+    sparse : bool
+        Whether or not the MGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MGXS is merged from one or more other MGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+
+    """
+
+    def __init__(self, domain=None, domain_type=None,
+                 groups=None, by_nuclide=False, name=''):
+        super(ChiPrompt, self).__init__(domain, domain_type, groups, by_nuclide, name)
+        self._rxn_type = 'chi-prompt'
+
+    @property
+    def scores(self):
+        return ['delayed-nu-fission', 'delayed-nu-fission',
+                'nu-fission', 'nu-fission']
+
+    @property
+    def filters(self):
+        # Create the non-domain specific Filters for the Tallies
+        group_edges = self.energy_groups.group_edges
+        energyout = openmc.Filter('energyout', group_edges)
+        energyin = openmc.Filter('energy', [group_edges[0], group_edges[-1]])
+        return [[energyin], [energyout], [energyin], [energyout]]
+
+    @property
+    def tally_keys(self):
+        return ['delayed-nu-fission-in', 'delayed-nu-fission-out',
+                'nu-fission-in', 'nu-fission-out']
+
+    @property
+    def rxn_rate_tally(self):
+        if self._rxn_rate_tally is None:
+            self._rxn_rate_tally = self.tallies['nu-fission-out'] - \
+                                   self.tallies['delayed-nu-fission-out']
+            self._rxn_rate_tally.sparse = self.sparse
+        return self._rxn_rate_tally
+
+    @property
+    def xs_tally(self):
+
+        if self._xs_tally is None:
+            delayed_nu_fission_in = self.tallies['delayed-nu-fission-in']
+            nu_fission_in = self.tallies['nu-fission-in']
+            prompt_nu_fission_in = nu_fission_in - delayed_nu_fission_in
+
+            # Remove coarse energy filter to keep it out of tally arithmetic
+            energy_filter = prompt_nu_fission_in.find_filter('energy')
+            prompt_nu_fission_in.remove_filter(energy_filter)
+
+            # Compute chi
+            self._xs_tally = self.rxn_rate_tally / prompt_nu_fission_in
+            super(ChiPrompt, self)._compute_xs()
+
+            # Add the coarse energy filter back to the nu-fission tally
+            prompt_nu_fission_in.filters.append(energy_filter)
+
+        return self._xs_tally
+
+    def get_slice(self, nuclides=[], groups=[]):
```

Comment:
It's too bad some of the specialized methods can't be inherited from `Chi`, but I see the challenge of designing the code to make that possible.